code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -642,11 +610,12 @@
 	var/entry_types = subtypesof(/datum/spellbook_entry) - /datum/spellbook_entry/item - /datum/spellbook_entry/summon - /datum/spellbook_entry/loadout
 	for(var/T in entry_types)
 		var/datum/spellbook_entry/E = new T
-		if(E.IsSpellAvailable())
-			entries |= E
-			categories |= E.category
-		else
+		if(GAMEMODE_IS_WIZARD && E.is_ragin_restricted)
 			qdel(E)
```

Comment:
Can most likely use the same construct as:
https://github.com/ParadiseSS13/Paradise/blob/73e8781b0ad9c51a5593868e0f9330d38e7e47a5/code/game/machinery/PDApainter.dm#L35-L37
`thing` here is a type. This way you don't have to create a new `spellbook_entry` and remove it again if it doesn't fit

Reply:
I tried doing 
```js
for(var/T in entry_types)
		var/datum/spellbook_entry/E = T
		if(GAMEMODE_IS_WIZARD && E.is_ragin_restricted)
			continue
		E = new T
        ...
```
But my spellbook ends up completely blank. I assume byond won't allow me to do the `E = new T` line.